src/features/vote/Vote.tsx

```diff
@@ -74,7 +74,7 @@ const Vote: FC<Props> = ({
   const {
     data: revealedVotes = [] as VoteRevealed[],
     refetch: refetchVoteRevealedEvents,
-  } = useVotesRevealedEvents(votingContract, votingAddress);
+  } = useVotesRevealedEventsRound(votingContract, votingAddress || hotAddress, roundId);
```

Comment:
two issues here. this was only working correctly with a 2 key, as votingAddress was soemtimes undefined when not using 2key. second the round id was not used, so you would be quering all revealed events, when we only cared about those for this round.

Reply:
Wow, this is a big fix. +1